### DISCUSSION TITLE
Im using int64 for primary key and its throwing me this error

### DISCUSSION THREAD

**Discussion by jeeva2334** [milvus-io/milvus#26459] [Q&A and General discussion]
from pymilvus import (
    connections,
    utility,
    FieldSchema,
    CollectionSchema,
    DataType,
    Collection,
)
import numpy

try:
    connections.connect("default", host="localhost", port="19530")
    print("connected Successfully")
except Exception as e:
    print(f"error occured: {e}")

book_id = FieldSchema(
    name="book_id", 
    dtype=DataType.INT64,
    is_primary=True, 
)

word_count = FieldSchema(
    name="book_id",
    dtype = DataType.INT64
)

book_intro = FieldSchema(
    name="book_id",
    dtype = DataType.FLOAT_VECTOR,
    dim=2
)

schema = CollectionSchema(
    fields=[book_id, word_count, book_intro],
    description="Test Book search"
)

collection_name = "book"

check = utility.has_collection("book")
print(check)

collection = Collection("book")
print(f"Collection {collection}")

error:

connected Successfully
Traceback (most recent call last):
  File "C:\codes\melvus\melvus connection\main.py", line 35, in <module>
    schema = CollectionSchema(
             ^^^^^^^^^^^^^^^^^
  File "C:\Users\Sanjay\AppData\Local\Programs\Python\Python311\Lib\site-packages\pymilvus\orm\schema.py", line 79, in __init__
    self._check_fields()
  File "C:\Users\Sanjay\AppData\Local\Programs\Python\Python311\Lib\site-packages\pymilvus\orm\schema.py", line 122, in _check_fields
    validate_primary_key(self._primary_field)
  File "C:\Users\Sanjay\AppData\Local\Programs\Python\Python311\Lib\site-packages\pymilvus\orm\schema.py", line 45, in validate_primary_key
    raise PrimaryKeyException(message=ExceptionsMessage.PrimaryKeyType)
pymilvus.exceptions.PrimaryKeyException: <PrimaryKeyException: (code=1, message=Primary key type must be DataType.INT64 or DataType.VARCHAR.)>

**Comment by yhmo** [comment]
All of the FieldSchema have the same name "book_id", which is not allowed.
Try this script:
```

from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility,
)

HOST = '127.0.0.1'
PORT = '19530'

collection_name = "test_abc"


try:
    connections.connect("default", host="localhost", port="19530")
    print("connected Successfully")
except Exception as e:
    print(f"error occured: {e}")

book_id = FieldSchema(
    name="book_id",
    dtype=DataType.INT64,
    is_primary=True,
)

word_count = FieldSchema(
    name="word_count",
    dtype = DataType.INT64
)

book_intro = FieldSchema(
    name="book_intro",
    dtype = DataType.FLOAT_VECTOR,
    dim=2
)

schema = CollectionSchema(
    fields=[book_id, word_count, book_intro],
    description="Test Book search"
)

collection_name = "book"

check = utility.has_collection(collection_name)
print(check)

collection = Collection(name=collection_name, schema=schema)
print(f"Collection {collection}")

```
